src/jit/codegenarm64.cpp

```diff
@@ -5723,17 +5750,163 @@ void CodeGen::genLeaInstruction(GenTreeAddrMode *lea)
+            {
+            case GT_EQ:
+                jmpKind[0] = EJ_eq;   // branch or set when equal (and no NaN's)
+                jmpKind[1] = EJ_vs;   // branch or set when we have a NaN 
+                break;
+
+            case GT_NE:
+                jmpKind[0] = EJ_ne;   // branch or set when not equal (or have NaN's)
+                jmpKind[1] = EJ_NONE;
+                break;
+
+            case GT_LT:
+                jmpKind[0] = EJ_lt;   // branch or set when less than (or have NaN's)
+                jmpKind[1] = EJ_NONE;
+                break;
+                
+            case GT_LE:
+                jmpKind[0] = EJ_le;   // branch or set when less than or equal (or have NaN's)
+                jmpKind[1] = EJ_NONE;
+                break;
+
+            case GT_GT:
+                jmpKind[0] = EJ_hi;   // branch or set when greater than (or have NaN's)
+                jmpKind[1] = EJ_NONE;
+                break;
+
+            case GT_GE:
+                jmpKind[0] = EJ_hs;   // branch or set when greater than or equal (or have NaN's)
+                jmpKind[1] = EJ_NONE;
+                break;
+
+            default:
+                unreached();
+            }
+        }
+        else  // ((cmpTree->gtFlags & GTF_RELOP_NAN_UN) == 0)
+        {
+            // Do not branch if we have an NaN, unordered
+            switch (cmpTree->gtOper)
+            {
+            case GT_EQ:
+                jmpKind[0] = EJ_eq;   // branch or set when equal (and no NaN's)
+                jmpKind[1] = EJ_NONE;
+                break;
+
+            case GT_NE:
+                jmpKind[0] = EJ_gt;   // branch or set when greater than (and no NaN's)
+                jmpKind[1] = EJ_lo;   // branch or set when less than (and no NaN's)
+                break;
+
+            case GT_LT:
+                jmpKind[0] = EJ_lo;   // branch or set when less than (and no NaN's)
+                jmpKind[1] = EJ_NONE;
+                break;
+
+            case GT_LE:
+                jmpKind[0] = EJ_ls;   // branch or set when less than or equal (and no NaN's)
+                jmpKind[1] = EJ_NONE;
+                break;
+
+            case GT_GT:
+                jmpKind[0] = EJ_gt;   // branch or set when greater than (and no NaN's)
+                jmpKind[1] = EJ_NONE;
+                break;
+
+            case GT_GE:
+                jmpKind[0] = EJ_ge;   // branch or set when greater than or equal (and no NaN's)
+                jmpKind[1] = EJ_NONE;
+                break;
+
+            default:
+                unreached();
+            }
+        }
+    }
+}
+
 // Generate code to materialize a condition into a register
 // (the condition codes must already have been appropriately set)
 
 void CodeGen::genSetRegToCond(regNumber dstReg, GenTreePtr tree)
 {
-    // Get the "jmpKind" using the gtOper kind
-    // Note that whether it is an unsigned cmp is governed by the GTF_UNSIGNED flags
+    emitJumpKind jumpKind[2];
```

Comment:
(since you are at it) Standard function header for this routine and for the one present in amd64